### ISSUE TITLE
Bulk Insert Missing Fields does not respect fieldsOrder in fileClass

### ISSUE THREAD

**Issue by ALegendsTale** [mdelobelle/metadatamenu#618]
Steps to reproduce issue:
1. Create new fileClass
2. Add mulitple fields
3. Change order of fields
4. Bulk insert missing fields into new file
5. Fields are inserted in order created instead of modified order

Release Version: 0.8.3

**Comment by Truncated** [comment]
Is the fieldsOrder **_intended_** to apply to the order of the properties in the target notes, or only in the order shown in the FileClass Fields configuration views?

I'd _**like it to**_, but I can't find any evidence that it is intended to.

**Comment by ChristinWhite** [comment]
Yep, I'm constantly moving things up and down in the frontmatter view, I'd love to have better ordering by default.

**Comment by AP2020C** [comment]
I also noticed the same problem. Very annoying. It slows me down. Actually it would be nice if, when reordering the props in the class, it would be asked whether to reorder them also in the files that implement the class, at least in those that have no other fields.
I also wish that some fields could be marked as required, and report error if it is absent.

### Steps to reproduce
Install MetadataMenu
Set File class Path in options
Make a new class
Add the fields: Name, Surname
Here problem 1
Rearrange fields putting Name on top
Create a new file and set the class to it
Insert field in frontmatter
Edit the class adding the field Middle and order: Name, Middle, Surname
Make a new file and assign the class to it
Here problem 2

### Expected result
When I add a new field to the class I want it to go to the bottom, when I add new fields to the frontmatter I want them to be listed in the same order as I set up the class

### Actual result
Problem 1: I have to reorder the fields, usually by adding a new field I want it to go at the end and not at the top

Problem 2: by adding a field after a file has already been created with the class (or maybe always, I haven't tried) the new fields are always added at the end.

**Comment by janosmathe** [comment]
I really think that this would an important feature implemented as a dedicated button available in the "Fields of..." view of an instance, which would best effort order the existing fields in the note according to the class definition, i.e.:
- all existing fields of the note that are in the class definition sorted according to the the order defined by the base ancestor's order, then it's descendant's order, all the way to the current class's order
- all other fields would be appended according to their order defined by the note

This would allow the updating of class instances if the class (or any ancestor) definitions get updated.

For defining a complete order over properties, a rendered list of class defined and inherited properties in `Fileclass fields` view would be even more amazing.

Similar efforts include:
- https://github.com/fchastanet/obsidian-metaflow/
- https://obsidian.rocks/quick-tip-organizing-properties-with-obsidian-linter/
